### DISCUSSION TITLE
Transaction manager on a cloud environment like EC2

### DISCUSSION THREAD

**Discussion by jjansenvr** [frankframework/frankframework#7097] [General]
What are the best practices on using transaction manager

Options
- BTM using a Samba share
- Narayana using a local folder
- Narayana using a Samba share
- Narayana using a Database

**Comment by nielsm5** [comment]
You may use either BTM or Narayana using a transaction logfile (on disk). It's up to the implementer to choose where to store this file. We do not recommend nor support smb shares (natively). 

Only Narayana can directly store it's log in the database.


The best practice depends on your environment. In your case EC2, I can only recommend using the database storage (in combination with Narayana). Unless you want to store the transaction log on disk, on a separate volume mount (not share!!). Which could work but becomes tricky with blue/green deployments, an/or restarts as you will need to write scripts to un- and re-mount the volumes to the correct EC2 instances.

**Reply by Laurens-makel** [reply]
Can the transaction logs and application logs be on the same RDS instance? 

I'd assume that if the RDS is multi-az deployed it would be possible but what about fail-overs when the master instance goes down for whatever reason, could be because of planned maintenance or the datacenter burning down? Can the transaction manager handle the situation that a transaction participant and its own transaction logs are not available at the same time, even for a short moment when the DNS change occurs?
